vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -801,6 +803,11 @@ public void setOpened(boolean opened) {
             ensureAttached();
         }
         setModality(opened && isModal());
+        // This part makes sure that the dialog gets detached if it gets closed
+        // but is set inert before it can get detached.
+        if (!opened) {
+            ElementUtil.setIgnoreParentInert(getElement(), true);
```

Comment:
I have a concern that the current approach allows unintentionally making the dialog ignore the parent inert when not supposed. For example, when you call `setOpened(false)` on a closed dialog.

Reply:
I added `isOpened()` to the if check to ensure that this line is only executed when an open dialog is being closed.